src/app/App.tsx

```diff
@@ -2,29 +2,44 @@ import { useState } from 'react';
 import classes from './App.module.css';
 import AppHeader from './components/AppHeader/AppHeader';
 import Feed from './components/Feed/Feed';
-import type { MatchClient } from '../types';
+import type { MatchClient, Profile } from '../types';
 import MatchDetails from './components/MatchDetails/MatchDetails';
 import User from './components/User/User';
+import useMatches from './hooks/useMatches';
 
 function App(): JSX.Element {
   const [match, setMatch] = useState<MatchClient | null>(null);
+  const [username, setUsername] = useState<Profile['username']>(null);
+  const matches = useMatches(username);
+
+  function handleButtonClick() {
```

Comment:
If you add more buttons, you are forced to rename this function. I would rename it now to understand which button was clicked:

```suggestion
  function handleBackToFeedClick() {
```